packages/webdriverio/src/commands/element/scrollIntoView.ts

```diff
@@ -61,55 +61,50 @@ export async function scrollIntoView (
 
     const deltaByOption = {
-        start: { y: elemRect.y, x: elemRect.x },
+        start: { y: elemRect.y - elemRect.height, x: elemRect.x - elemRect.width },
         center: { y: elemRect.y - Math.round((viewport.height - elemRect.height) / 2), x: elemRect.x - Math.round((viewport.width - elemRect.width) / 2) },
-        end: { y: elemRect.y - (viewport.height - elemRect.height - 20), x: elemRect.x - (viewport.width - elemRect.width) }
+        end: { y: elemRect.y - (viewport.height - elemRect.height), x: elemRect.x - (viewport.width - elemRect.width) }
     }
+
     let [deltaX, deltaY] = [deltaByOption.start.x, deltaByOption.start.y]
-    if (options && typeof options !== 'boolean') {
+    if (options === true) {
+        options = { block: 'start', inline: 'nearest' }
+    }
+    if (options === false) {
+        options = { block: 'end', inline: 'nearest' }
+    }
+    if (options && typeof options === 'object') {
         const { block, inline } = options
         if (block === 'nearest') {
-            const nearestDistance = Math.min(...Object.values(deltaByOption).map(delta => Math.abs(delta.y)))
-            deltaY = Object.values(deltaByOption).find(delta => Math.abs(delta.y) === nearestDistance)!.y
+            const nearestYDistance = Math.min(...Object.values(deltaByOption).map(delta => delta.y))
+            deltaY = Object.values(deltaByOption).find(delta => delta.y === nearestYDistance)!.y
         } else if (block) {
             deltaY = deltaByOption[block].y
         }
         if (inline === 'nearest') {
-            const nearestDistance = Math.min(...Object.values(deltaByOption).map(delta => Math.abs(delta.x)))
-            deltaX = Object.values(deltaByOption).find(delta => Math.abs(delta.x) === nearestDistance)!.x
+            const nearestXDistance = Math.min(...Object.values(deltaByOption).map(delta => delta.x))
+            deltaX = Object.values(deltaByOption).find(delta => delta.x === nearestXDistance)!.x
         } else if (inline) {
             deltaX = deltaByOption[inline].x
         }
     }
 
-    if (origin) {
-        deltaX = deltaX > viewport.width ? Math.round(viewport.width - elemRect.width - (elemRect.width / 2) - 30) : Math.round(deltaX)
-        deltaY = deltaY > viewport.height ? Math.round(viewport.height - elemRect.height - (elemRect.height / 2) - 30) : Math.round(deltaY)
-
-        windowX = 0
-        windowY = 0
-    } else {
-        deltaX = Math.round(deltaX - origScrollX)
-        deltaY = Math.round(deltaY - origScrollY)
-
-        windowX = Math.round(windowX)
-        windowY = Math.round(windowY)
-    }
+    // take into account the current scroll position
+    deltaX = Math.round(deltaX - scrollX)
+    deltaY = Math.round(deltaY - scrollY)
 
     try {
         return await browser.action('wheel')
-            .scroll({ duration: 200, x: windowX, y: windowY, deltaX, deltaY, origin })
+            .scroll({ duration: 0, x: deltaX, deltaY, origin: this })
```

Comment:
Hello, @christian-bromann, @erwinheitzman 
Could you please tell me what's the reason for using `x: deltaX` here?
Why can't we just use `deltaX`, `deltaY` instead?

Reply:
I think I made when changing this, it should be y: deltaY

Reply:
i see. So it will be changed in future releases?
I just didn't want to create a bug for this minor thing. Scroll is working for me, but with warning and fallback to web api

Reply:
Yeah I can correct this, thing is it won't show up in the release notes because we use the issues to generate the release notes.

Regarding the warning, it falls back to the native web scrollIntoView when the webdriver approach throws an error so it would be interesting to find out why it's failing at that attempt and with what error.

Reply:
got it. Thanks